src/styles/common/StyledButton.ts

```diff
@@ -0,0 +1,190 @@
+import styled, { css } from 'styled-components'
+import { variants } from 'styled-theming'
+
+import ColourSheet from '../colours'
+import StyleSheet from '../styles'
+
+const {
+  white,
+  whiteDark,
+  blue,
+  blueDark,
+  successLight,
+  successDark,
+  warningLight,
+  warningDark,
+  dangerLight,
+  dangerDark,
+  bgLight,
+  bgDark,
+} = ColourSheet
+
+const { buttonBorder } = StyleSheet
+
+// Used in stories
+// Good to keep around altough not required
+export enum ButtonVariations {
+  default,
+  primary,
+  secondary,
+  danger,
+  success,
+  warning,
+  cancel,
+  theme,
+}
+
+// Create our variated Button Theme
+// 'kind' refers to a prop on button
+// <ButtonBase kind="danger" />
+export const ButtonTheme = variants<'kind', keyof typeof ButtonVariations>('mode', 'kind', {
```

Comment:
You know, now that I think about it, enums aren't there so that we use `keyof typeof enum`, they are to be used directly
`kind: ButtonVariations ---> <Button kind={ButtonVariations.primary}/>`
If we use strings, we should sticktostrings `type ButtonVariations = 'default' | 'primary' | ...`

Reply:
Yeah i fucked this up it should be 
```ts
enum thing {
  button = 'button',
...
}
```